OOP Python/teacher_and_student.py

```diff
@@ -0,0 +1,132 @@
+class Human:
+    def __init__(self, name, surname, age):
+        self.name = name
+        self.surname = surname
+        self.age = age
+    
+    
+    def get_full_name(self):
+        full_name = f"{self.name} {self.surname}"
+        return full_name
+    
+    
+    def set_full_name(self,full_name):
+        self.name, self.surname = full_name.split(" ")
+        
+    
+    def __repr__(self):
+        return self.get_full_name()
+
+
+class Student(Human):
+    def __init__(self, name, surname, age, marks):
+        super().__init__(name, surname, age)
+        
+        self.marks = marks
+        
+    def get_average_mark(self):
+        return sum(self.marks) // len(self.marks)
+    
+    
+    def get_min_mark(self):    
+        return min(self.marks)
+    
+    
+    def get_max_mark(self):    
+        return max(self.marks)
+
+
+class Teacher:
+    def __init__(self,students):
+        self.students = students
+
+
+    def get_list_of_names_by_average_mark(self):
+        return (sorted(self.students,
+                       key=lambda student: student.get_average_mark(),
+                       reverse=True))
+
+
+    def get_student_by_name(self, name):
+        for student in self.students:
+            if student.name == name:
+                return student
+        return None
+
+
+    def remove_student_by_name(self, name):
+        for student in self.students:
+            if student.name == name:
+                return self.students.remove(student)
+        return None
+
+
+    def update_student_by_name(self, replace_student, name):
+        for student in self.students:
+            if student.name == name:
+                self.students.remove(student)
+                return self.students.append(replace_student)
+        return None
+
+
+    def find_cheater_student(self):
+        for student in self.students:
+            avg_real = sum(student.marks) // len(student.marks)
+            avg_curent = student.get_average_mark()
+            if avg_curent != avg_real:
+                return f"{student.get_full_name()}"
+        return None
+
+
+class FakeStudent(Student):
+    def __init__(self, name, surname, age, marks):
+        super().__init__(name, surname, age, marks)
+
+        self.__cheated_marks = self.cheat()
+
+    def cheat(self):
+        return [min(mark * 2, 10) for mark in self.marks]
+
```

Comment:
cheat має бути приватним методом (__cheat), який не може визиватись ззовні (з екземпляру). Він одразу визивається при створенні екземпляру ( в __init)
```suggestion
class FakeStudent(Student):
    def __init__(self, name, surname, age, marks):
        super().__init__(name, surname, age, marks)

        self.__cheated_marks = self.__cheat()

    def __cheat(self):
        return [min(mark * 2, 10) for mark in self.marks]

```